libgalois/include/katana/PropertyGraph.h

```diff
@@ -262,38 +262,51 @@ class KATANA_EXPORT PropertyGraph {
     rdg_.set_mirror_nodes(std::move(a));
   }
 
+  /// Return the node property table for local nodes
+  const std::shared_ptr<arrow::Table>& node_properties() const {
+    return rdg_.node_properties();
+  }
+
+  /// Return the edge property table for local edges
+  const std::shared_ptr<arrow::Table>& edge_properties() const {
+    return rdg_.edge_properties();
+  }
+
 public:
   /// PropertyView provides a uniform interface when you don't need to
   /// distinguish operating on edge or node properties
   struct PropertyView {
     PropertyGraph* g;
+    const PropertyGraph* const_g;
```

Comment:
Why is this needed? I suspect there is a detail of C++ typing I don't understand.

Reply:
Sorry, I asked about this in #1479, which is the associated enterprise PR.  Here are the questions.

In PropertyGraph.h, to implement const and non-const PropertyViews I have a PropertyGraph* called g and a const PropertyGraph* called const_g. Is there a better way of doing this?

For the const PropertyView, I leave several function pointers as null. I could make functions that assert, but it seemed annoying to have several different types of dummy functions that do nothing but assert.

Reply:
I see no connection between the two pointer types and the nulls vs asserts. But it's no big deal.

Reply:
I really am interested if there is a better way to do what I want without `g` and `const_g`.

I included the second question because it seemed like the kind of thing you might know or have an opinion on and I didn't want to make you read #1479.

Reply:
`g` and `const_g` point to the same object? and `const_cast` doesn't work or a function that wraps `const_cast`?

Reply:
They point to the same object.
Can you explain your question about `const_cast`?  The problem comes from creating a `PropertyView `and a `const PropertyView`.

Reply:
I see.

One pattern would be to use a template that captures the const or lack of const-ness, e.g.,

```cpp
using ConstView = View<const PropertyGraph*>;
using MutableView = View<PropertyGraph*>;
```

Though given that the const view actually has less functions (the nullptr member functions), I'd suggest just making two classes and skip the templates, e.g.,
```cpp
class ReadOnlyView {

};

class MutableView: public ReadOnlyView {
public:
  Result<void> AddProperties(....) {....}
};

ReadOnlyView edge_property_view() const {....}
MutableView edge_property_view() {...}
```

whether you want to make `Mutable` and `ReadOnlyView` virtual base classes and have concrete classes for nodes and edges or want to continue using member function pointers is up to you.

I originally used member function pointers because I saw (and still see) PropertyGraph as a closed ecosystem, so how we implemented the required (limited) polymorphism was an implementation detail, so I picked what I thought was the more compact way. Virtual functions are fine too.

Reply:
I have taken your advice and made two explicitly named structs.  I included the read-only version in the mutable version because it led to less duplication.  

I introduced explicit names for the "constructors" instead of const/non-const because I found myself casting a `DistPropertyGraph `pointer to a const pointer and invoking `edge_property_view()` which seemed like an unnecessarily obscure mechanism.

Reply:
Thank you!